modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridMapQueryExecutor.java

```diff
@@ -637,6 +647,9 @@ public void onCacheStop(String cacheName) {
         private final AtomicReferenceArray<QueryResult> results;
 
         /** */
+        private final AtomicReferenceArray<AtomicReference<GridAbsClosure>> cancels;
```

Comment:
Looks complex, why not just preallocate AtomicReference<GridAbsClosure>[] array?